auth/auth.go

```diff
@@ -320,20 +321,29 @@ func (c *baseClient) verifyIDToken(ctx context.Context, idToken string, checkRev
 		}
 	}
 
-	if c.isEmulator || checkRevoked {
-		revoked, err := c.checkRevoked(ctx, decoded)
+	if c.isEmulator || checkRevokedOrDisabled {
+		errMessageDisabled := &internal.FirebaseError{
+			ErrorCode: internal.InvalidArgument,
+			String:    "User has been disabled",
+			Ext: map[string]interface{}{
+				authErrorCode: userDisabled,
+			},
+		}
+		err := c.checkRevokedOrDisabled(ctx, decoded, userDisabled, errMessageDisabled)
 		if err != nil {
 			return nil, err
 		}
 
-		if revoked {
-			return nil, &internal.FirebaseError{
-				ErrorCode: internal.InvalidArgument,
-				String:    "ID token has been revoked",
-				Ext: map[string]interface{}{
-					authErrorCode: idTokenRevoked,
-				},
-			}
+		errMessageRevoked := &internal.FirebaseError{
+			ErrorCode: internal.InvalidArgument,
+			String:    "ID token has been revoked",
+			Ext: map[string]interface{}{
+				authErrorCode: idTokenRevoked,
+			},
+		}
+		err = c.checkRevokedOrDisabled(ctx, decoded, idTokenRevoked, errMessageRevoked)
```

Comment:
This seems to be calling `checkRevokedOrDisabled()` twice, which is not correct. That results in extra RPCs that we don't need. See my comment on how to refactor the `checkRevokedOrDisabled()` function, so this can be avoided.